gto-support-kotlin-coroutines/src/main/kotlin/org/ccci/gto/android/common/kotlin/coroutines/ReadWriteMutex.kt

```diff
@@ -19,23 +19,22 @@ fun ReadWriteMutex(): ReadWriteMutex = ReadWriteMutexImpl()
 
 @VisibleForTesting
 internal class ReadWriteMutexImpl : ReadWriteMutex {
-    private val stateMutex = Mutex()
-    private val readerOwner = Any()
+    private val readLockMutex = Mutex()
     @VisibleForTesting
     internal val readers = AtomicLong(0)
 
     override val write = Mutex()
     override val read = object : Mutex {
         override suspend fun lock(owner: Any?) {
-            stateMutex.withLock {
+            readLockMutex.withLock {
                 while (true) {
                     val count = readers.get()
                     check(count < Long.MAX_VALUE) {
                         "Attempt to lock the read mutex more than ${Long.MAX_VALUE} times concurrently"
                     }
-                    if (count == 0L) write.lock(readerOwner)
+                    if (count == 0L) write.lock()
```

Comment:
this had been throwing an `IllegalStateException` occasionally because we were trying to lock the write lock with the `readerOwner` mutex before the unlock on line 48 completed.